central/processlisteningonport/datastore/datastore_impl.go

```diff
@@ -646,3 +646,89 @@ func (ds *datastoreImpl) RemovePLOPsWithoutProcessIndicatorOrProcessInfo(ctx con
 
 	return int64(len(plopsToDelete)), nil
 }
+
+// Removes PLOPs without poduids between a range of ids.
+func (ds *datastoreImpl) RemovePLOPsWithoutPodUIDOnePage(ctx context.Context, prevId string, nextId string) (int64, error) {
+	ds.mutex.Lock()
+	defer ds.mutex.Unlock()
+
+	query := fmt.Sprintf(deletePLOPsWithoutPoduidInPage, prevId, nextId)
+	commandTag, err := ds.pool.Exec(ctx, query)
+
+	if err != nil {
+		return 0, err
+	}
+
+	return commandTag.RowsAffected(), nil
+}
+
+// Given a set of rows with ids, returns the id of the last row. This is useful for pagination.
+func (ds *datastoreImpl) getLastIdFromRows(ctx context.Context, rows pgx.Rows) (string, error) {
+	id := ""
+
+	for rows.Next() {
+		if err := rows.Scan(&id); err != nil {
+			return "", pgutils.ErrNilIfNoRows(err)
+		}
+	}
+
+	return id, nil
+}
+
+// Given an id and a limit, returns the id a limit number of rows after the given id. This is useful
+// for efficient pagination.
+func (ds *datastoreImpl) GetNextPageId(ctx context.Context, prevId string, limit int) (string, error) {
+	ds.mutex.Lock()
+	defer ds.mutex.Unlock()
+
+	query := fmt.Sprintf(getLastIdFromPage, prevId, limit)
+	rows, err := ds.pool.Query(ctx, query)
+
+	if err != nil {
+		// Do not be alarmed if the error is simply NoRows
+		err = pgutils.ErrNilIfNoRows(err)
+		if err != nil {
+			log.Warnf("%s: %s", query, err)
+		}
+		return "", err
+	}
+	defer rows.Close()
+
+	nextId, err := ds.getLastIdFromRows(ctx, rows)
+
+	if err != nil {
+		return "", err
+	}
+
+	return nextId, nil
```

Comment:
```suggestion
	return nextId, rows.Err()
```
Some errors will be missed unless we check `rows.Err()`

Reply:
I accepted the other suggestion instead of this one.